src/routes/issue-trigger-rule/update.ts

```diff
@@ -0,0 +1,40 @@
+import Joi from 'joi';
+import { ControllerGeneratorOptionsWithClient } from '../../comms/controller';
+import { IssueTriggerRule } from '../../models/issue-trigger-rule';
+
+type HashId = string;
+
+type Request = {
+  params: {
+    hashId: HashId
+  }
+  body: Omit<IssueTriggerRule, 'clientHashId' | 'deviceTypeHashId'>
+}
+type EffectiveRequest = Request;
+
+type Response = void
+
+const JoiHashId = Joi.string().required().example('xd2rd4');
+
+const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
+  method: 'put',
+  path: '/:hashId',
+  params: Joi.object().keys({
+    hashId: JoiHashId,
+  }).required(),
+  body: Joi.object().keys({
+    priorityLevel: Joi.number().integer().valid(1, 2).required()
+      .example('1'),
+    missedReports: Joi.number().integer().required().example('12'),
+    offlineForSeconds: Joi.number().integer().required().example('1800'),
```

Comment:
Typically we make the properties in the update body optional (so that the client doesn't have to provide the full object). Could you do that here as well?

Not wrong per se, but why is updating the deviceTypeHashId not allowed?

Reply:
> Typically we make the properties in the update body optional (so that the client doesn't have to provide the full object).

I followed the REST spec, PUT needs all fields, PATCH doesn't. But PATCH is not allowed here

> Could you do that here as well?

Yes 👍 

> Not wrong per se, but why is updating the deviceTypeHashId not allowed?

I don't know yet how the creation of a rule at the `deviceType` level is gonna look in the monitoring environment as it was not really discussed. But I would expect a view where we list the `deviceTypes`, and when clicked we see its rules.
Not a list of rules where we can select the `deviceType`.
So updating the `deviceType` for a rule did not make sense to me (it would be a "transfer/copy" of rule really). But doing this opens a lot more questions:
Is there already a rule for this `deviceType` with the same `priority` ? Does the user want to override or abort ?
Is it a transfer or a copy/duplication of rule ?

Reply:
What are your thoughts on this, @Arinono?

Reply:
@everhardt I like your suggestion. Actually this where I went with yesterday with my latest commit. But I need to run some extra tests to make sure I did it right

Reply:
👆 https://github.com/withthegrid/platform-sdk/pull/126/commits/524886e40e035acdf8e02a7b974e5bea9bc9db7c#diff-917cd19227b7cd206abc3f72187da9de6deae23775aa2d8fd12735236279a230R7-R26

Reply:
And about the hashId vs. deviceType/priority discussion?

Reply:
oooh, maybe I miss understood something.
You suggest that a rule is "identified" by the deviceType/priority "combo" instead of hashId ?

If yes, I didn't consider/understood it because I never seen something like this before 🤔 

I'd have to look at some SQL docs to see/remember how to do it properly on the platform side

Reply:
> You suggest that a rule is "identified" by the deviceType/priority "combo" instead of hashId ?

Yes indeed, as we're not allowing more than 1 for the combination. A unique index on the two would be the solution in MySQL

Reply:
Ok, I'll try to make the changes by the end of the day (both sdk and platform) so you can have a look

Reply:
Last question, on this unique index (`UNIQUE KEY` ?), should I also include the `client_id` ?

> `UNIQUE KEY/INDEX rule (client_id, device_type_id, priority_level)` ?

Reply:
Yes :)!